src/setup_node_env/harden.js

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var hook = require('require-in-the-middle');
+var create = require('object-prototype').create;
+
+// Ensure `process.env` doesn't inherit from `Object.prototype`. This gives
+// partial protection against similar RCE vulnerabilities as described in
+// CVE-2019-7609
+process.env = Object.assign(create(), process.env);
```

Comment:
Doing this causes the following error to be thrown when trying to use `exec` (and likely other methods):

```
TypeError: Cannot convert object to primitive value
    at normalizeSpawnArguments (child_process.js:529:31)
    at spawn (child_process.js:545:16)
    at Object.execFile (child_process.js:221:15)
    at exec (child_process.js:152:18)
```

Node.js [iterates over all of the keys in `process.env`](https://github.com/nodejs/node/blob/84eec80f84e9752dd07e293c7aa3d852918f3350/lib/child_process.js#L526-L531), and `object-prototype` is defining all of it's functions as enumerable.

Changing

```
for (const name of functions) {
  ObjectPrototype[name] = prototypelessFunction(Object.prototype[name])
}
```

to
```
for (const name of functions) {
  const originalDescriptor = Object.getOwnPropertyDescriptor(Object.prototype, name);
  Object.defineProperty(ObjectPrototype, name, {
    ...originalDescriptor,
    value: prototypelessFunction(Object.prototype[name])
  });
}
```

fixes this issue.

Reply:
Fixed 👍